zero_bin/rpc/src/native/txn.rs

```diff
@@ -1,31 +1,41 @@
-use std::collections::{HashMap, HashSet};
+use std::{
+    collections::{HashMap, HashSet},
+    sync::OnceLock,
+};
 
 use __compat_primitive_types::{H256, U256};
 use alloy::{
-    primitives::{keccak256, Address, B256},
+    primitives::{keccak256, Address, B256, U160},
     providers::{
         ext::DebugApi as _,
-        network::{eip2718::Encodable2718, Ethereum, Network},
+        network::{eip2718::Encodable2718, Ethereum, Network, TransactionResponse},
         Provider,
     },
     rpc::types::{
-        eth::Transaction,
-        eth::{AccessList, Block},
+        eth::{AccessList, Block, Transaction},
         trace::geth::{
-            AccountState, DiffMode, GethDebugBuiltInTracerType, GethTrace, PreStateConfig,
+            AccountState, DefaultFrame, DiffMode, GethDebugBuiltInTracerType, GethDebugTracerType,
+            GethDebugTracingOptions, GethDefaultTracingOptions, GethTrace, PreStateConfig,
             PreStateFrame, PreStateMode,
         },
-        trace::geth::{GethDebugTracerType, GethDebugTracingOptions},
     },
     transports::Transport,
 };
 use anyhow::Context as _;
+use evm_arithmetization::{CodeDb, JumpDestTableWitness};
 use futures::stream::{FuturesOrdered, TryStreamExt};
 use trace_decoder::{ContractCodeUsage, TxnInfo, TxnMeta, TxnTrace};
 
-use super::CodeDb;
 use crate::Compat;
 
+/// Provides a way to check in constant time if an address points to a
+/// precompile.
+fn precompiles() -> &'static HashSet<Address> {
+    static PRECOMPILES: OnceLock<HashSet<Address>> = OnceLock::new();
+    PRECOMPILES
+        .get_or_init(|| HashSet::<Address>::from_iter((0..9).map(|x| Address::from(U160::from(x)))))
```

Comment:
You're missing `0x9` (Blake) and `0xa` (KZG).

Reply:
Thanks!

Reply:
This isn't resolved yet, you're still including the 0 address

Reply:
You are right.